tests/std/tests/P0896R4_counted_iterator_death/test.cpp

```diff
@@ -16,6 +16,8 @@ int otherArray[5]{10, 20, 30, 40, 50};
 vector<int> checkedArray{10, 20, 30, 40, 50};
 using vit = vector<int>::iterator;
 
+static_assert(input_iterator<vit>); // TRANSITION, FIXME (File an issue to investigate this later?)

```

Comment:
Its not stupid if it works!

Reply:
I think the recursion is a consequence of permissive mode + `using namespace std` allowing the ADL lookup for `iter_move(some_counted_iterator)`  inside the implementation of `ranges::iter_move` to find `ranges::iter_move`. I'm investigating yanking `using namespace std` from the tests and have stumbled across what looks like a bug in placeholder types with *type-constraint*s (`concept auto foo = bar;`).

Reply:
Ditto can't merge `FIXME`.